v2/router.go

```diff
@@ -48,9 +48,38 @@ type Procedure struct {
 	Signature string
 }
 
+// EncodingProcedure specifies a single encoding-level handler and is mapped to a TransportProcedure by the router.
```

Comment:
We don't necessarily need this here, but a description for how the `EncodingProcedure` coordinates with the `TransportProcedure` would be helpful. Like @mh-park already described, this is still a little fuzzy for me too.